server_spec.rb

```diff
@@ -0,0 +1,234 @@
+require_relative 'server'
+require_relative 'spec_helper'
+
+describe Server do
+  let(:app) { described_class.new }
+  before :all do
+    @db = SQLite3::Database.new('test.db')
+  end
+
+  after :all do
+    File.delete('test.db')
+  end
+
+  def create_tables(db)
+    db.execute(<<-SQL
+      create table messages (
+             id varchar(255),
+             to_user varchar(255),
+             from_user varchar(255),
+             message varchar(255),
+             status varchar(255));
+             SQL
+              )
+    db.execute(<<-SQL
+      create table users(
+             id varchar(255));
+             SQL
+              )
+  end
+
+  def seed_for_create_message(db, to_id, from_id)
+    db.execute(<<-SQL
+      insert into users(id)
+      values
+        ("#{to_id}"),
+        ("#{from_id}");
+        SQL
+              )
+  end
+
+  def seed_for_get_status(db, id, status)
+    db.execute(<<-SQL
+      insert into messages(id, to_user, from_user, message, status)
+      values('#{id}', '#{SecureRandom.uuid}', '#{SecureRandom.uuid}', 'this is a test', '#{status}');
+        SQL
+              )
+  end
+
+  def clear_db(db)
+    db.execute(<<-SQL
+      drop table users;
+      SQL
+              )
+    db.execute(<<-SQL
+      drop table messages;
+      SQL
+              )
+  end
+
+  def message_record_with?(db:, id:, attribute:, value:)
+    row = db.execute(<<-SQL
+        SELECT #{attribute}, from_user FROM messages WHERE id ='#{id}'
+        SQL
+    )
+    row.any? && row.flatten.first == value
+  end
+
+  def no_message_record?(db)
+    db.execute(<<-SQL
+        SELECT * FROM messages
+        SQL
+              ).none?
+  end
+
+  describe 'POST messages' do
+    let(:to_id) { SecureRandom.uuid }
+    let(:from_id) { SecureRandom.uuid }
+    let(:attributes) do
+      {
+        to: to_id,
+        from: from_id,
+        message: 'this is a test'
+      }
+    end
+    before do
+      create_tables(@db)
+      seed_for_create_message(@db, to_id, from_id)
+      allow(described_class::ExternalApiClient).to receive(:send_message).and_return(true)
+    end
+
+    after do
+      clear_db(@db)
+    end
+
+
+    subject(:response) { post "/messages/#{format}", request_body }
+
+    context 'json' do
+      let(:format) { 'json' }
```

Comment:
Given how similar our json and xml tests suites will look, could we extract the shared logic as 'shared examples'?